### PR TITLE
Added lenient flag for synonym token filter (6.x)

### PR DESCRIPTION
**Pull request by sohaibiftikhar**
Backport for #31484 

(cherry picked from 88c270d8443c573cf593a3d7d429be5bdad14001)

### CODE REVIEW

**Comment by imotov** [comment]
@elasticmachine add to whitelist

**Comment by cbuescher** [comment]
@mayya-sharipova since you reviewed #31484, do you think this needs another review or is it good to merge once CI passed?

**Review by mayya-sharipova — APPROVED** [review]

**Pull request merged**
